src/main/java/seedu/mtracker/MTracker.java

```diff
@@ -15,7 +18,16 @@ public MTracker() {
 
     public void run() {
         TextUi.greetAtStartUp();
-
+        String userInput = InputParser.getUserInput();
+        String[] components = InputParser.getCommandComponents(userInput);
+        try {
+            Command c = InputParser.filterByCommandType(components);
+            c.execute();
+        } catch (InvalidCommandError e) {
+            System.out.println(e.getMessage());
+        } catch (Exception e) {
```

Comment:
Need only one catch block. Cause Exception is superclass of InvalidCommandError. So when invalidCommandError is thrown, it will also be caught and errorMessage shown would be the over-ridden invalidCommandError

Reply:
With this in mind, cause the filterByCommandType may throw exception, the command object may not ever be initialised so in the catch block initialise the command to a new InvalidCommand
^ I leave the implementation details for this up to you

Reply:
ok

Reply:
Assigning a new InvalidCommand in the catch block will be done after putting everything inside while loop when fixing issue #29 .